WordPress/src/main/java/org/wordpress/android/util/wizard/WizardManager.kt

```diff
@@ -0,0 +1,52 @@
+package org.wordpress.android.util.wizard
+
+import android.arch.lifecycle.LiveData
+import org.wordpress.android.util.Event
+import org.wordpress.android.viewmodel.SingleLiveEvent
+
+class WizardManager<T>(
+    private val steps: List<T>
+) {
+    private val _navigatorLiveData = SingleLiveEvent<T>()
+    val navigatorLiveData: LiveData<T> = _navigatorLiveData
+    private var currentStepIndex: Int = -1
+
+    fun showNextStep() {
+        if (isIndexValid(++currentStepIndex)) {
+            _navigatorLiveData.value = steps[currentStepIndex]
+        } else {
+            throw IllegalStateException("Invalid index.")
+        }
+    }
+
+    /**
+     * Fragments need to inflate the view in OnCreateView only when getView() != null, otherwise their state
+     * gets lost.
+     */
+    fun onBackPressed() {
+        --currentStepIndex
+    }
+
+    fun hasNextStep(): Boolean {
+        return isIndexValid(currentStepIndex + 1)
+    }
+
+    private fun isIndexValid(currentStepIndex: Int): Boolean {
+        return currentStepIndex >= 0 && currentStepIndex < steps.size
+    }
+}
+
+/**
+ * Marker interface representing a single step/screen in a wizard
+ */
+interface WizardStep
```

Comment:
I am only mentioning this because we talked about this on the call. An alternative approach to keeping an index might be to use an approach similar to a LinkedList or a tree. Linked list approach might look something like this:

```
interface WizardStep {
    val previous: WizardStep?
    val next: WizardStep?
}
```

In this ^ case, `WizardManager` would take the initial step instead of the list. These changes would allow us to get rid of index tracking and although unlikely, dealing with out of bounds indexes.

_I have a sneaking suspicion that there might be a nice way to tie this to the communication between view models such as `onVerticalSelected` without actually exposing that it's a step to them._

Reply:
👍 Thanks for writing this down. I've decided to think about it for a few more days as it's not urgent and the current solution works.

Reply:
Sounds good!